index.php

```diff
@@ -38,14 +38,24 @@
 $languages = array();
 if ($handle = opendir('lang')) {
     while (false !== ($entry = readdir($handle))) {
-        if ($entry != "." && $entry != "..") {
-             array_push($languages, str_replace(".inc.php", "", $entry));
+        if ($entry != "." && $entry != ".." ) {
+            $entry_lang = str_replace(".inc.php", "", $entry);
+            # Only add language to possibilities if it is the default language or part of the allowed languages 
+            # empty $allowed_lang <=> all languages are allowed
+            if ($entry_lang == $lang || empty($allowed_lang) || in_array($entry_lang, $allowed_lang) ) {
+                array_push($languages, $entry_lang);
+            }
         }
     }
     closedir($handle);
 }
 $lang = detectLanguage($lang, $languages);
-require_once("lang/$lang.inc.php");
+if (file_exists("lang/$lang.inc.php")) {
```

Comment:
@plewin you are right for error reporting, it should happen before the lang selection.

And yes with error reporting, we will know if the lang configuration is good or not. The failback to the default lang is already coded in the detectLanguage, I agree we should not force a failback on English after.